project/hychiu02/README.md

```diff
@@ -0,0 +1,96 @@
+# AutoEncoder for arrhythmia detection
+
+## Basic information
+
+[GitHub](https://github.com/hychiu02/AutoEncoder-for-arrhythmia-detection)
+
+Arrhythmia dectection using ECG signal
+
+## Problem to solve
+Heart diseases are one of the significant reasons for death all over the planet. To perform arrhythmia detection, Electrocardiograms(ECGs) is used to monitor hte functioning of heart by capturing electrical activity. Because of needs of real-time inference, the model sizeand  parameters should be small, autoencoder seems to be a good choice. I want to develop a method to build, train and inference a MLP model with C++. 
+
+## Prospective Users
+People who want to build Multilayer Perceptron model for particular case.
+
+## System Architecture
+There will be a Node class for storing weight, a Layer class with Nodes for forward and backward weights calculation and a MLP class containing Layers  
+The flow will look like this
+![](Flow.png)
+
+
+## API Description
+- Dataset class
+```
+ Dataset
+  the constructor initialize Dataset
+  - parameters:
+    dataPath: ECG file path

```

Comment:
Do you require all data to be loaded to memory?

Reply:
Actually I plan to load all file names to memory, and when model need data, a part of file will be load.

Reply:
Loading data on the fly takes quite some code development and it is hard to have a robust implementation on different platform.  Imagine someone runs your code on NFS.

It's OK.  Just to keep it in mind that I/O is hard.

Reply:
Sorry i didn't think well. I'll try to survey how to run across different platform. Currently I just have rough idea to run on my device.

Reply:
Blind spots are not something needing to apologize for.  Please do not say sorry for that or it will be hard for the discussions to go productively.